### PR TITLE
feat: search optimisation based on skip pointers

### PR DESCRIPTION
**Pull request by savannahar68**
## What does this PR do?
- Using skip pointers in search Operation of logs to enhance performance for multi word search

## Refer to a related PR or issue link
- [(Related PR or issue)](https://github.com/infinohq/infino/issues/20)

## Checklist

- [X]  I have written the necessary rustdoc comments.
- [X]  I have added the necessary unit tests and integration tests. (for non-documentation changes)
- [] More tests

High level sudo code

```
Iterate over posting list 

    if acc != len(acc) 

        If acc < initial_value -> acc ++
        If acc > initial value
            if acc < initial_value + 1 -> Scan the current posting_block and add matching acc (or subsequent acc's) to temp_acc_lsit
            else -> initial_value + 1
        If acc == inital value -> Add acc in temp_acc_lsit | acc ++

    End of accumulator or End of current posting list iteration -> set acc = temp_acc_lsit

```

### CODE REVIEW

**Comment by savannahar68** [comment]
TODO: add more tests and compare against current benchmark

**Comment by savannahar68** [comment]
@vinaykakade please take a look at logic.

Some duplicate code is present, will try to refactor the code

File: src/main.rs [hunk 1]
```diff
@@ -298,7 +298,7 @@ async fn append_ts(
 ) -> Result<(), (StatusCode, String)> {
   debug!("Appending time series entry: {:?}", ts_json);
 
-  let is_queue = if state.queue.is_some() { true } else { false };
```
Comment (vinaykakade) [inline src/main.rs:301]: thanks for fixing this!

**Comment by savannahar68** [comment]
Perf improvement with skip pointer without runtime de-compressing of posting block 
image

**Comment by savannahar68** [comment]
Closing this PR against #82

**Pull request closed without merging**